edb/pgsql/compiler/relgen.py

```diff
@@ -2556,7 +2556,7 @@ def process_set_as_agg_expr(
                 ctx=xctx)
 
     if wrapper:
-        stmt = wrapper
+        ctx.rel = stmt = wrapper
```

Comment:
This seems fishy.  `wrapper` may come from a `ctx.subrel()` so we're doing some weird inversion here.  Why is this needed?

Reply:
When a aggregate function with an initial_val was used at the top-level of a query (without a SelectStmt wrapping it, which happens for fragments like the USING clauses of SET TYPE), the actual results were in one of the rvars in the returned SetRVars but they weren't actually making it into the result rel.

This is because in those cases, we were totally ignoring the input stmt in favor of creating a new statement, which was dubious. This fix was even more dodgy, and I've pushed a cleanup.

(Though it also feels off to me that `_process_toplevel_query` doesn't do anything with the rvars in the common case. But trying to `_include_rvars` there causes us to try to generate an infinite amount of SQL)